### PR TITLE
Add initial documentation files

### LINKED ISSUE #22: Add initial documentation for FEniCS plug in
- [x] Design the documentation home page
- [x] Link the documentation home page in the CUQIpy-FEniCS plugin readme file
- [x] Documentation for API
- [x] Documentation for installation (remove information from README about Docker+Conda and about google co-lab and put it as a part of installation page or section in the documentation.). The page can be similar to read me but with additional installation details. (related issues: #13 #27 ) (this item will be considered in a separate issue #67)

### PR DESCRIPTION
**Pull request by amal-ghamdi**
closes #22

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: docs
+
+on:
+  push:
+    branches:
+      - add_documentation
```
Comment (amal-ghamdi) [inline .github/workflows/docs.yml:6]:
```suggestion
      - main
```
Using add_documentation branch is temp for testing

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          pip install -r requirements-dev.txt
+
+      - name: Sphinx build
+        shell: bash -l {0}
+        run: |
+          conda activate fenicsproject
+          sphinx-build docs ./public
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/add_documentation' }}
```
Comment (amal-ghamdi) [inline .github/workflows/docs.yml:60]:
```suggestion
        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```
Using add_documentation branch is temp for testing (to publish the doc now)

**Review by jakobsj — APPROVED** [review]
Great job adding this. I have taken a quick look and it appears to work nicely and provide valuable documentation. I agree it can be merged, and further improved along the way, for example by adding small example calls to supplement the raw api overview.

**Review by nabriis — APPROVED** [review]
Thanks @amal-ghamdi! Looking really good. I approve, but could you add the new custom CSS from CUQIpy to have the panel work nicely in dark mode?

https://github.com/CUQI-DTU/CUQIpy/blob/main/docs/_static/custom.css

**Comment by amal-ghamdi** [comment]
Thank you @jakobsj and @nabriis for your review! 

I added the new css sheet. I also captured some of the remaining items in a new issue #67 including Jakob's suggestion. Merging :)

**Pull request merged**